### ISSUE TITLE
json_to_sheet should have an option to write down only columns, specified in headers option

### ISSUE THREAD

**Issue by Lonli-Lokli** [SheetJS/sheetjs#2455]
```
  const worksheet = XLSX.utils.json_to_sheet(datasource, {
          header: ['A']
        });
```

I was going to see only A column in the table, now it's just written as first and all others are listed later.
Can you introduce an option to exclude columns which are not listed in headers option?

**Comment by reviewher** [comment]
You can preprocess the data source to remove unnecessary data:

```js
const headers = ["A"];
const aoo = datasource.map(obj => Object.fromEntries(headers.map(h => ([h, obj[h]])));
const worksheet = XLSX.utils.json_to_sheet(aoo, {
  header: headers
});
```

For broader compatibility, this can be done with simple array ops:

```js
var aoo = datasource.map(function(obj) { 
  return headers.reduce(function(acc, key) {
    acc[key] = obj[key];
    return acc;
  }, ({}));
});
```

**Comment by Lonli-Lokli** [comment]
@reviewher I don't agree with closing issue. From my point of view it should be part of feature libs, of course I can workaroind it. But it will take more time & memory to process.

**Comment by reviewher** [comment]
What should this option be called?

How would it work if headers are not specified?  Will it merely pull the headers from the first row?

**Comment by Lonli-Lokli** [comment]
For the simplicity
1. As a proposal, it can be named 'whitelist': boolean
2. It should be ignored if no headers specified
3. Default value for backward compatibility should be false
